src/Core/Commit.js

```diff
@@ -0,0 +1,385 @@
+ * @param {Element} element
+ * @param {List} children
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitChildren (element, children, host, signature, mode) {
+	var length = children.length
+	var next = children.next
+
+	while (length-- > 0) {
+		commitMount(next, element, element, host, signature, mode)
+		next = next.next
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitMount (element, sibling, parent, host, signature, mode) {
+	element.host = host
+	element.parent = parent
+	element.context = host.context
+
+ 	switch (element.id) {
+ 		case SharedElementComponent:
+ 			element.work = SharedWorkTask
+ 			
+ 			commitMount(componentMount(element), sibling, parent, element, signature, mode)
+ 			element.DOM = getHostChildren(element).DOM
+
+ 			if (element.ref)
+ 				commitReference(element, element.ref, SharedReferenceAssign)
+ 			
+ 			if (element.owner[SharedComponentDidMount])
+ 				getLifecycleMount(element, SharedComponentDidMount)
+
+ 			element.work = SharedWorkSync
+ 			return
+ 		case SharedElementPortal:
+ 			element.DOM = DOM(element.type)
+ 			break
+ 		case SharedElementPromise:
+ 			commitPromise(element, element)
+ 		case SharedElementFragment:
+ 			element.DOM = DOM(getDOMNode(parent))
+ 			break
+ 		case SharedElementNode:
+ 			element.xmlns = getDOMType(element, parent.xmlns)
+ 		case SharedElementText:
+ 			switch (mode) {
+ 				case SharedMountClone:
+ 					if (element.DOM = commitQuery(element, parent, element.id === SharedElementText))
+	 					break
+ 				default:
+ 					element.DOM = commitCreate(element)
+ 					
+ 					if (signature < SharedMountInsert)
+ 						commitAppend(element, parent)
+ 					else
+ 						commitInsert(element, sibling, parent)
+ 			}
+
+ 			if (element.id === SharedElementText)
+ 				return
+ 	}
+
+ 	commitChildren(element, element.children, host, SharedMountAppend, mode)
+ 	commitProperties(element, element.props, SharedPropsMount)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ * @return {(Promise|void)}
+ */
+function commitUnmount (element, parent, signature) {
+	if (signature > SharedElementEmpty)
+		commitDismount(element, signature)
+
+	if (element.id !== SharedElementComponent)
+		return commitRemove(element, parent)
+
+	if (queue > 0)
+		return Promise.all(stack)
+			.then(commitResolve(element, parent, signature))
+			.catch(commitResolve(element, parent, SharedErrorPassive))
+
+	commitUnmount(getHostChildren(element), parent, SharedElementEmpty)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ * @return {function}
+ */
+function commitResolve (element, parent, signature) {
+	return function (e) {
+		commitUnmount(getHostChildren(element), parent, queue = stack.length = SharedElementEmpty)
```

Comment:
This should probably clear the queue outside of the promise context, after `Promise.all`